Framework/API/inc/MantidAPI/IndexTypeProperty.h

```diff
@@ -0,0 +1,61 @@
+#ifndef MANTID_API_INDEXTYPEPROPERTY_H_
+#define MANTID_API_INDEXTYPEPROPERTY_H_
+
+#include "MantidAPI/DllConfig.h"
+#include <MantidKernel/PropertyWithValue.h>
+
+namespace Mantid {
+namespace API {
+
+enum IndexType { SpectrumNum = 1, WorkspaceIndex = 2 };
+
+/** IndexTypeProperty : TODO : Add support for DetectorIDs
```

Comment:
The DetectorID support will not be added now because we need to develop a method for translating from DetectorID to SpectrumIndexSet. Simon has pushed this down the priority list since there are not many cases where this is used compared to spectrum numbers and workspace indices.

Maybe `IndexPropertyConversionType`? (Seems like a lot to type). I guess what we want to convey is that the user will receive a `SpectrumIndexSet` after converting from this selected type. We also want to establish a relationship with the `IndexProperty`.